src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs

```diff
@@ -56,7 +56,7 @@ private async Task<IOperationResult> DoBulkInsert(Action<IOperationProgress> onP
                             long totalSize = 0;
                             while (true)
                             {
-                                var task = parser.MoveNext(docsCtx);
+                                var task = parser.MoveNext(docsCtx, new BlittableMetadataModifier(docsCtx));
```

Comment:
can we reuse the modifier here?

Reply:
The modifier is bounded to the context. 
I wanted to avoid complexity for now, since we switch the context here (few times).

Reply:
BlittableMetadataModifier needs to be disposed. We need to detect if context has switched.

Reply:
The `BlittableMetadataModifier` is disposed when the context is and the behavior is not different from that we had before this change.
However we do need to do that properly, so I opened an issue for that:
https://issues.hibernatingrhinos.com/issue/RavenDB-13445